### DISCUSSION TITLE
Renaming of EnumMeniItem

### DISCUSSION THREAD

**Discussion by nickloris** [davetcc/tcMenu#515] [Q&A] [answered]
Hi!
is it possible to rename the list of items in runtime?
is there also the possibility to add more item as well in runtime?

Thanks
Nicola

**Comment by davetcc** [comment]
Yes there are two options for this, the best way by far would be to use a scroll choice item. This option is fully documented here:

https://tcmenu.github.io/documentation/arduino-libraries/tc-menu/menu-item-types/scrollchoice-menu-item/
 
—-

The other way that is not recommended would be to create an enum item yourself at runtime and set its info block to RAM.

**Comment by davetcc** [comment, marked as answer]
Although this is from some time ago, I'll document all the ways here:

* The best way to do this by far is scroll choice menu item as described above. It is fully integrated into designer - https://tcmenu.github.io/documentation/arduino-libraries/tc-menu/menu-item-types/scrollchoice-menu-item/
* The second best way is to use an enum menu item marked to have its info block in RAM. This means that the name, and entries are stored in RAM and can therefore be changed. However, it is far more difficult than the above.
